source/postprocess/visualization.cc

```diff
@@ -326,10 +330,12 @@ namespace aspect
       if (std::isnan(last_output_time))
         {
           last_output_time = this->get_time() - output_interval;
+          last_output_timestep = this->get_timestep_number();
         }
 
       // return if graphical output is not requested at this time
       if ((this->get_time() < last_output_time + output_interval)
+          && (this->get_timestep_number() < last_output_timestep + maximum_timesteps_between_outputs)
```

Comment:
this doesn't work -- `maximum_timesteps_between_outputs` may be the largest integer around if the user did not specifically specify it, so the addition on the right is always going to wrap around. I suspect this condition to never be true. 

You'll have to write the condition in a way that doesn't overflow/wraps around, or limit the values the parameter can have to something so that "in all cases we can think of", there is no wrap around. For example, if we make sure the parameter is never larger than one billion, and we can reasonably assume that nobody ever runs a simulation with a billion time steps, then that would work because for integers, wrap around only happens at two billions.

Reply:
Funnily enough, it will work as the tests show (although by accident, I did not think of your point when reviewing). The reason it works is that the default value is set to `std::numeric_limits<int>::max()`, but the value is stored in an `unsigned int`, so the one sign-bit (allowing 4 billion as max value, instead of the 2 billion of int) is saving us from an integer overflow, as long as nobody is running a model for more than 2 billion timesteps (a safe assumption). Are you fine with the rest of the PR?

Reply:
No, actually, the value is stored as a `double` as you remarked above. If it was stored as an `unsigned int`, that would work along the same lines as I mentioned above. In any case, though, I think this line deserves a comment that explains the issue with overflow, and in line 740, the pattern should be written as `Patterns::Integer(0,std::numeric_limits<int>::max())` to make clear that the value is not allowed to be larger than half the unsigned range.